### PR TITLE
added typings for the panel plugin.

### PR DESCRIPTION
**Pull request by nkappler**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

~If adding a new definition:~

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://codemirror.net/doc/manual.html#addon_panel
- [x] ~Increase the version number in the header if appropriate.~
- [x] ~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.~

~If removing a declaration:~

### CODE REVIEW

**Comment by nkappler** [comment]
@mihailik @Pr1st0n

File: types/codemirror/codemirror-panel.d.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// See docs https://codemirror.net/doc/manual.html#addon_panel
+
+import * as CodeMirror from "codemirror";
+
+declare module "codemirror" {
+    interface Panel {
+        /**Removes the panel from the editor */
+        clear: () => void;
+        /**Notifies panel that height of DOM node has changed */
+        changed: (height?: number) => void;
```
Comment (mihailik) [inline types/codemirror/codemirror-panel.d.ts]: It's easier to read with method-like notation, no? Can we lose those extra arrow characters please?

```typescript
interface Panel {
  clear(): void;
  changed(height?: number): void;
}

File: types/codemirror/codemirror-panel.d.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    interface Editor {
+
+        /**
+         * Places a DOM node above or below an editor and shrinks the editor to make room for the node.
+         * When using the `after`, `before` or `replace` options, if the panel doesn't exists or has been removed, the value of the `position` option will be used as a fallback.
+         * @param node the DOM node
+         * @param options optional options object
+         */
+        addPanel: (node: HTMLElement, options?: ShowPanelOptions) => Panel;
```
Comment (mihailik) [inline types/codemirror/codemirror-panel.d.ts]: Here as well, please.

```typescript
addPanel(node: HTMLElement, options?: ShowPanelOptions): Panel;
```

**Review by mihailik — CHANGES_REQUESTED** [review]
Thanks, great work closing the gap between official docs and typings!

It would be great to tweak it just tiny bit, for smoother neater readability, if thats OK.

Many thanks!

**Review by mihailik — APPROVED** [review]
Awesome, thanks a lot @nkappler, appreciate extra effort there!

**Pull request merged**